src/RollbarClientSubmitter.test.mts

```diff
@@ -411,7 +414,7 @@ describe(`Class: ${RollbarClientSubmitter.name}`, () => {
           const testMessage = 'test message';
           submitter.report('error', testMessage);
 
-          const actualPayload = JSON.parse(navigator.sendBeacon.mock.calls[0][1]);
+          const actualPayload = JSON.parse(navigator.sendBeacon.mock.calls[0][1]) as IPayload;
```

Comment:
This is the only line **in the code** with a `tsc` error remaining. I can't get `tsc` or `eslint` to treat `sendBeacon` as a mocked function, so I'm a little stumped here.

Reply:
I ended up ignoring these errors due to not being able to get to the bottom of this problem in a timely manner. As you can see, I've added a `TODO` comment to keep an eye on this so when this issue is fixed in the future (or maybe I'm doing something wrong?) I'll remember to address it.

Reply:
Unfortunately these are many of the things I've tried. I had to add the `@ts-ignore` to make the `tsc` complaints go away. I tried both of your suggestions and neither worked unfortunately. I'm not sure what else to try.

Reply:
Tried it myself to see.

We apparently can't do my first idea, as it's not allowed to override a global 😅 

Though the second idea does work:

```typescript
type MockSendBeacon = SpyInstance<Parameters<Navigator["sendBeacon"]>, ReturnType<Navigator["sendBeacon"]>>;
const sendBeaconAsMock = navigator.sendBeacon as unknown as MockSendBeacon;
const actualPayload = JSON.parse(sendBeaconAsMock.mock.calls[0][1]) as IPayload;
```

(ps: this still complains that what we are trying to pass to `JSON.parse` might not be a string as it is typed as `data?: BodyInit | null | undefined`, this sounds like maybe something that should be double checked then whether it's alright to expect it to be a `string` given it shouldn't have such requirement.

Reply:
Awesome! This works great, I added `as string` to the parameter being passed to `JSON.parse()` and now zero errors. I still think Vitest should handle this correctly, but thank you again for helping me get to the bottom of this!